File: src/main/java/org/app/infrastructure/endpoints/dto/GetFollowingListDTO.java
package org.app.infrastructure.endpoints.dto;

import org.app.domain.events.FollowingEvent;

import java.util.List;

public class GetFollowingListDTO {
    public List<GetUserItemDTO> following;
    public int page;
    public int limit;
    public int totalPage;

    public GetFollowingListDTO(List<GetUserItemDTO> following, int page, int limit, int totalPage) {
        this.following = following;
        this.page = page;
        this.limit = limit;
        this.totalPage = totalPage;
    }

    public GetFollowingListDTO(FollowingEvent event) {
        this.following = event.getFollowings().stream().map(GetUserItemDTO::new).toList();
        this.page = event.getPage();
        this.limit = event.getLimit();
        this.totalPage = event.getTotalPage();
    }

    public static GetFollowingListDTO from(FollowingEvent event){
        return new GetFollowingListDTO(event);
    }
}

File: go/app/main/infrastructure/endpoints/dto/get_following_list_dto.go
package dto

import (
    "mutualAttentionSystem/app/main/domain/events"
    "mutualAttentionSystem/app/main/infrastructure/utils"
)

type GetFollowingListDTO struct {
    Followings []*GetUserItemDTO
    Page       int
    Limit      int
    TotalPage  int
}

func ToGetFollowingListDTO(event *events.GetUserEvent, page int, limit int) (*GetFollowingListDTO, error) {
    pageable, err := utils.NewPableable(page, limit, len(event.User.GetFollowings()))
    if err != nil {
        return nil, err
    }
    getFollowingListDTO := GetFollowingListDTO{make([]*GetUserItemDTO, 0), page, limit, pageable.TotalPage}
    for _, user := range event.User.GetFollowings() {
        getFollowingListDTO.Followings = append(getFollowingListDTO.Followings, ToGetUserItem(user))
    }

    return &getFollowingListDTO, nil
}
